### ISSUE TITLE
Parameter validation failed error with autoscaling describe-launch-configurations with --next-token argument

### ISSUE THREAD

**Issue by robbwagoner** [aws/aws-cli#1299]
### AWS CLI version 1.7.23

```shell
$ aws --version
aws-cli/1.7.23 Python/2.7.6 Darwin/14.1.0
```

### Observed Results

Using the `autoscaling describe-launch-configurations` subcommand with a limit of 20 launch configs per call:

```shell
$ aws autoscaling describe-launch-configurations --region us-east-1 --max-items 20
{
    "NextToken": "None___20",
    "LaunchConfigurations": [
        {
            "UserData":
...
}
```

Using both arguments `--max-items` and `--next-token`, to retrieve the second batch of results generates an error:

```shell
$ aws autoscaling describe-launch-configurations --max-items 20 --next-token None___20

Parameter validation failed:
Unknown parameter in input: "max_items", must be one of: LaunchConfigurationNames, NextToken, MaxRecords
```

When `--max-items` argument is removed, the next-token value is considered invalid:
```shell
 $ aws autoscaling describe-launch-configurations --next-token "None___20"

A client error (InvalidNextToken) occurred when calling the DescribeLaunchConfigurations operation: The token 'None___20' was invalid
```

### Expected Results

Using `--max-items 20 --next-token Non___20` would result in receiving Launch Configurations 21-40 in the output of the AWS CLI.

**Comment by kyleknap** [comment]
@robbwagoner 

It looks like you are using the autoscaling's specific pagination API parameters (--next-token) for the request you are making. In the CLI, if a command can be paginated, we hide those parameters and expose our own parameters to control pagination. We prefer users to use the parameters that we expose (--starting-token) because the tokens that we generate for you in the CLI will not be valid if you pass it directly to the service's API, as noted by the error message you got back. If you use the pagination parameters the CLI exposes, we will take the token, process it, and use the appropriate token when we make the request to autoscaling.

So in short try the following:
```
$ aws autoscaling describe-launch-configurations --max-items 20 --starting-token None___20
```

Note the use of ``--starting-token`` instead of ``--next-token``.

Let me know how that works out for you.

**Comment by robbwagoner** [comment]
Thanks @kyleknap. Transposing the argument name was the problem. Using `--starting-token` resolved the matter. Closing.
